src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -587,11 +590,11 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
 
   @inline private def lazySlidingImpl(size: Int, step: Int): Iterator[LazyList[A]] =
     if (knownIsEmpty) Iterator.empty
-    else Iterator.empty ++ slidingImpl(size, step) // concat with empty iterator so that `slidingImpl` is lazy
+    else Iterator.empty ++ slidingImpl(size, step, size - step max 0) // concat with empty iterator so that `slidingImpl` is lazy
 
-  private def slidingImpl(size: Int, step: Int): Iterator[LazyList[A]] =
-    if (isEmpty) Iterator.empty
-    else Iterator.single(take(size)) ++ drop(step).slidingImpl(size, step)
+  private def slidingImpl(size: Int, step: Int, minLen: Int): Iterator[LazyList[A]] =
+    if (lengthLteq(minLen)) Iterator.empty
```

Comment:
need to handle first round differently, because first can be arbitrarily small (though not empty)

Reply:
oops, this was meant as a TODO not a description